### PR TITLE
fix: fix react query function timeout and retries, minor ui bugs

### PR DESCRIPTION
**Pull request by poomthiti**

### CODE REVIEW

File: src/lib/query-utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const createQueryFnWithTimeout =
+  <T>(queryFn: () => Promise<T>, ms: number = 6 * 60 * 1000) =>
```
Comment (evilpeach) [inline src/lib/query-utils/index.ts:2]: That should be 6 * 1000 ms

File: src/lib/pages/account-details/components/common.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { CustomIcon } from "lib/components/icon";
+
+export const ErrorFetching = () => (
```
Comment (songwongtp) [inline src/lib/pages/account-details/components/common.tsx:3]: ErrorFetching.tsx ?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/services/accountService.ts [hunk 3]
```diff
@@ -2,7 +2,9 @@ import { useWallet } from "@cosmos-kit/react";
 import type { UseQueryResult } from "@tanstack/react-query";
 import { useQuery } from "@tanstack/react-query";
 
-import type { Addr, Balance } from "lib/types";
+import { useCelatoneApp, useChainId } from "lib/app-provider";
+import { getAccountIdByAddressQueryDocument } from "lib/query/account";
```
Comment (evilpeach) [inline src/lib/services/accountService.ts:6]: Please import from `lib/query`

File: src/lib/services/accountService.ts [hunk 4]
```diff
@@ -24,6 +26,27 @@ export const useAccountBalances = (
...
+
+export const useAccountId = (walletAddress: Option<Addr>) => {
+  const { indexerGraphClient } = useCelatoneApp();
+  const chainId = useChainId();
+  const queryFn = () => {
+    if (!walletAddress)
+      throw new Error("Error fetching account id: failed to retrieve address.");
+    return indexerGraphClient
+      .request(getAccountIdByAddressQueryDocument, { address: walletAddress })
+      .then(({ accounts_by_pk }) => accounts_by_pk?.id);
```
Comment (evilpeach) [inline src/lib/services/accountService.ts:45]: add type

File: src/lib/services/accountService.ts [hunk 5]
```diff
@@ -24,6 +26,27 @@ export const useAccountBalances = (
...
+  };
+  return useQuery(["account_id", chainId, walletAddress], queryFn, {
```
Comment (evilpeach) [inline src/lib/services/accountService.ts:47]: add `indexerGraphClient` instead of `chainId`

File: src/lib/services/txService.ts [hunk 6]
```diff
@@ -136,21 +145,30 @@ export const useTxsByAddressPagination = (
       pageSize,
```
Comment (evilpeach) [inline src/lib/services/txService.ts:145]: accountId

File: src/lib/services/accountService.ts [hunk 7]
```diff
@@ -24,6 +26,27 @@ export const useAccountBalances = (
...
-    { enabled: !!currentChainRecord || !!walletAddress }
+    {
+      enabled: !!currentChainRecord || !!walletAddress,
+      retry: 1,
+      refetchOnWindowFocus: false,
+    }
   );
 };
+
+export const useAccountId = (walletAddress: Option<Addr>) => {
```
[the remark this answers is not in the document]
Comment (evilpeach) [reply src/lib/services/accountService.ts:37]: Also, add chain id

File: src/lib/services/expression/txExpression.ts [hunk 8]
```diff
@@ -34,20 +34,35 @@ const generateSearch = (search: string) =>
...
+            transaction: {
+              ...(hasFilter ? generateActionsFilter(filters) : {}),
+              ...(search
+                ? { _or: generateSearch(search.toLocaleLowerCase()) }
+                : {}),
+            },
+          }
+        : {}),
+    };
+  }, [address, accountId, filters, isSigner, search]);
```
Comment (evilpeach) [inline src/lib/services/expression/txExpression.ts:68]: refactor this to destructure pattern

**Pull request merged**